_posts/2019-07-25-testy-frontendu-okiem-full-stacka.md

```diff
@@ -0,0 +1,141 @@
+---
+layout:    post
+title:     "Testy jednostkowe frontendu okiem programisty full stack"
+published: true
+date:      2019-07-25 10:00:00 +0100
+author:    mmendlik
+tags:
+    - unit test
+---
+
+Niełatwo znaleźć wymówkę, żeby nie pisać testów jednostkowych. Obecność frameworków ułatwiających tę czynność w projektach, z którymi stykamy się na co dzień, nie powinna na żadnym chociaż trochę doświadczonym programiście robić wrażenia. Nie trzeba pewnie Ciebie przekonywać, że **jedne z wielu zalet pisania testów jednostkowych**, to:
+* zmuszenie twórcy do zastanowienia się nad zadaniem sprawdzanego kodu (co może potencjalnie poprawić architekturę aplikacji),
+* ułatwienie wczesnego wyłapywania błędów,
+* ekspozycja przypadków brzegowych,
+* ułatwienie zrozumienia działania kodu osobom, które go nie tworzyły.
+
+
+W wielu nowoczesnych aplikacjach internetowych, w tym np. we wnioskach Eximee, **duża część logiki znajduje się po stronie klienckiej**, hipokryzją byłoby pominięcie testów w tak istotnym elemencie aplikacji, bo przecież wiadomo, że najszybciej wychodzą na jaw błędy właśnie po stronie klienckiej. Niemniej nawet przy ogromnej liczbie narzędzi wspomagających proces pisania testów jednostkowych, programiści mogą mieć problem z **wyznaczeniem właśnie tych jednostek**.
```

Comment:
Powtórzenie "po stronie klienckiej". Nie brzmi najgorzej, ale chyba można to zdanie przearanżować, żeby brzmiało lepiej. Może coś na zasadzie "[..] bo przecież wiadomo, że błędy najszybciej wychodzą na jaw właśnie tam"

Reply:
done